compiler/semstmts.nim

```diff
@@ -680,6 +689,9 @@ proc semVarOrLet(c: PContext, n: PNode, symkind: TSymKind): PNode =
         addToVarSection(c, result, n, a)
         continue
       var v = semIdentDef(c, a[j], symkind, false)
+      if {sfThread, sfNoInit} * v.flags != {}:
+        a[^1] = c.graph.emptyNode
+        def = c.graph.emptyNode
```

Comment:
Seems to be a bad idea, what about `var x {.noInit.} = f()`? You turn this into `var x {.noInit.} = <default>`? Why?